.github/workflows/on-pull-request-commit.yml

```diff
@@ -0,0 +1,27 @@
+---
+name: PR commit
+
+on:
+  merge_group:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  main:
+    name: Build fork-off
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout git
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: Cardinal-Cryptography/github-actions/install-rust-toolchain@v6
+
+      - name: Build
+        run: |
+          cargo build --release
+
+      - name: Check
```

Comment:
I'd move things around, i.e. first check then build (as check is faster than build, doesn't generate binaries)